src/univariate/continuous/rayleigh.jl

```diff
@@ -83,7 +83,7 @@ function logccdf(d::Rayleigh, x::Real)
 end
 ccdf(d::Rayleigh, x::Real) = exp(logccdf(d, x))
 
-cdf(d::Rayleigh, x::Real) = -expm1(logccdf(d, x))
+cdf(d::Rayleigh, x::Real) = false - expm1(logccdf(d, x))  # zero (false) ensures that cdf cannot be -0
```

Comment:
I would prefer something that fixes the issue more directly - this seems like a quite weird workaround. The underlying problem here is that the implementation of `expm1` in base returns
```julia
julia> expm1(0.0)
0.0

julia> expm1(-0.0)
-0.0
```
Since
```julia
julia> iszero(-0.0)
true

julia> iszero(0.0)
true
```
both positive and negative zero are proper zeros, I'm not completely sure why a negative zero would be a problem here. Do you have a MWE where it causes problems?

If we want to change it maybe we could just use
```suggestion
# we ensure that the cdf cannot be -0
cdf(d::Rayleigh, x::Real) = max(- expm1(logccdf(d, x)), 0)
```

Reply:
See the linked issue for motivation. A simple example:
```
julia> using Distributions
julia> d = Rayleigh(1)
# nonzero works fine:
julia> rand(Binomial(100, cdf(d, 0.1)))
2
# zero doesn't:
julia> rand(Binomial(100, cdf(d, 0)))
ERROR: InexactError: trunc(Int64, -Inf)
<...>
# but it would, if cdf didn't return negative zero:
julia> rand(Binomial(100, 0.0))
0
```

Reply:
Actually, a simpler solution is possible. Changed.